addons/sale_coupon/models/sale_order.py

```diff
@@ -242,7 +242,7 @@ def _get_reward_values_percentage_amount(self, program):
             if line:
                 discount_line_amount = min(line.price_reduce * (program.discount_percentage / 100), amount_total)
                 if discount_line_amount:
-                    taxes = self.fiscal_position_id.map_tax(line.tax_id)
+                    taxes = self.fiscal_position_id.map_tax(line.tax_id.filtered(lambda t: t.amount_type != 'fixed'))
```

Comment:
Conceptual question: shouldn't we do the filtering after the fpos tax mapping ?
I dunno if it sometimes happens/makes sense to map non fixed taxes to fixed or the opposite :thinking:

Reply:
You are right, even if it seems weird it's possible to map fixed to non-fixed and vice-versa, so we should filter after mapping.